lib/openscap_parser/rule.rb

```diff
@@ -1,68 +1,62 @@
 # frozen_string_literal: true
 
 require 'openscap_parser/rule_identifier'
-require 'openscap_parser/rule_reference'
+require 'openscap_parser/rule_references'
+require 'openscap_parser/xml_file'
 
 # Mimics openscap-ruby Rule interface
 module OpenscapParser
   class Rule
-    def initialize(rule_xml: nil)
-      @rule_xml = rule_xml
+    include OpenscapParser::Util
+    include OpenscapParser::XmlFile
+    include OpenscapParser::RuleReferences
+
+    def initialize(parsed_xml: nil)
+      @parsed_xml = parsed_xml
     end
 
     def id
-      @id ||= @rule_xml['id']
+      @id ||= @parsed_xml['id']
     end
 
     def selected
-      @selected ||= @rule_xml['selected']
+      @selected ||= @parsed_xml['selected']
     end
 
     def severity
-      @severity ||= @rule_xml['severity']
+      @severity ||= @parsed_xml['severity']
     end
 
     def title
-      @title ||= @rule_xml.at_css('title') &&
-        @rule_xml.at_css('title').text
+      @title ||= @parsed_xml.at_css('title') &&
+        @parsed_xml.at_css('title').text
     end
 
     def description
       @description ||= newline_to_whitespace(
-        @rule_xml.at_css('description') &&
-          @rule_xml.at_css('description').text || ''
+        @parsed_xml.at_css('description') &&
+          @parsed_xml.at_css('description').text || ''
       )
     end
 
     def rationale
       @rationale ||= newline_to_whitespace(
-        @rule_xml.at_css('rationale') &&
-          @rule_xml.at_css('rationale').text || ''
+        @parsed_xml.at_css('rationale') &&
+          @parsed_xml.at_css('rationale').text || ''
       )
     end
 
-    def references
-      @references ||= reference_nodes.map do |node|
-        RuleReference.new(reference_xml: node)
-      end
-    end
-
-    def reference_nodes
-      @reference_nodes ||= @rule_xml.xpath('reference')
+    alias :rule_reference_nodes_old :rule_reference_nodes
```

Comment:
We're overriding https://github.com/dLobatog/openscap_parser/pull/18/files#diff-c883c147fe302a85453f4dc58d46c051R20.